File: config/config-sbc2ha/src/main/java/com/dfi/sbc2ha/config/sbc2ha/definition/platform/bus/ModbusBusConfig.java
package com.dfi.sbc2ha.config.sbc2ha.definition.platform.bus;

import com.dfi.sbc2ha.config.sbc2ha.definition.enums.PlatformType;
import com.dfi.sbc2ha.config.sbc2ha.definition.enums.UartType;
import com.fasterxml.jackson.annotation.JsonProperty;
import lombok.Data;

@Data
public class ModbusBusConfig extends BusConfig {
    public static final String BUS_ID = "modbus";

    UartType uart;
    @JsonProperty("use_diozero")
    boolean useDiozero = true;

    public ModbusBusConfig() {
        busId = BUS_ID;
        setPlatform(PlatformType.MODBUS);
    }
}

File: config/editor/src/app/definition/platform/bus/ModbusBusConfig.ts
import {BusConfig} from "./BusConfig";
import {UartType} from "../../enums/UartType";
import {PlatformType} from "../../enums/PlatformType";


export class ModbusBusConfig extends BusConfig {
    private static BUS_ID: string = "modbus";
    uart: UartType

    public constructor(uart: UartType) {
        super(PlatformType.MODBUS, ModbusBusConfig.BUS_ID);
        this.uart = uart;
    }
}
